File: Sorting/practiseSelectionSort.cpp
#include<bits/stdc++.h>
#include<string.h>
using namespace std;

void swap(string arr[], int x, int y){
	string temp = arr[x];
	arr[x] = arr[y];
	arr[y] = temp;
}
void selectionSort(string arr[], int n){
	int min_indx;
	
	for(int i=0; i<n; i++){
		min_indx = i;
		for(int j=i+1; j<n; j++){
			if(arr[j]<arr[min_indx]){
				min_indx = j;
			}
		}
		swap(arr,min_indx,i);
	}
}

int main(){
	
	string arr[] =  { "GeeksforGeeks", "Practice.GeeksforGeeks", "GeeksQuiz" };
	int n = sizeof(arr)/sizeof(arr[0]);
	
	selectionSort(arr,n);
	for(int i=0; i<n; i++){
		cout << arr[i] << endl;
	}

	
	return 0;
}

File: ALGORITHM in JavaScript/Sorting/PractiseSelectionSort.js
// Input  : paper true soap floppy flower
// Output : floppy, flower, paper, soap, true
function swap(arr,x,y) {
    let temp = arr[x];
    arr[x] = arr[y];
    arr[y] = temp
}
function sort(arr,n){
    var min_indx;
    for(let i=0; i<n; i++){
        min_indx = i;
        for(let j=i+1; j<n; j++){ // 2 1
            if(arr[j]<arr[min_indx]){
                min_indx = j;
            }
        }
        swap(arr,min_indx,i)
    }
}
// let arr = [ 'paper', 'true', 'soap', 'floppy', 'flower' ];
// let arr =  [ "GeeksforGeeks",
// "Practice.GeeksforGeeks",
// "GeeksQuiz" ];
let arr = ['b','a','e','k','c']
let n = arr.length;

sort(arr,n)
for(let i=0; i<n; i++){
    console.log(arr[i]);
}
